gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -230,6 +231,33 @@ class {{ service.name }}RestTransport({{ service.name }}Transport):
     {% endif %}
     {% endfor %}
 
+{# TODO: Allow client creation to explictly specify connection
+         parameters, such as secure/insecure connections, and have tests use
+         that. Then remove this function.
+         Tracking issue:
+         https://github.com/googleapis/gapic-generator-python/issues/901
+#}
+def protocol_for(host: str) -> str:
+    """Determine the URL protocol to use for connecting to the given host.
+
+    This function returns 'http' for hosts that match
+    `^localhost(:.*)?$` and `https` for all others. This is intended
+    to allow integration testing over an insecure connection to a
+    locally running server.
+
+    Args:
+        host (str): The hostname for REST connection.
+
+    Returns:
+        str: The protocol (`http` or `https`) to use for connecting to `host`.
+    """
+    
+    LOCALHOST='localhost'
+    LOCALHOST_LEN=len(LOCALHOST)
+    
+    if host.startswith(LOCALHOST) and (len(host)==LOCALHOST_LEN or host[LOCALHOST_LEN]==":"):
+        return 'http'
```

Comment:
I have a slight preference for regex match here instead (especially since the docstring already seems to describe this in terms of regex :)) https://docs.python.org/3/library/re.html